src/pymanopt/backends/backend.py

```diff
@@ -30,15 +31,23 @@ def inner(self, *args, **kwargs):
     return inner
 
 
+class DTypePrecision(Enum):
+    """Enum for the dtype precision of the backend."""
+
+    SINGLE = 32
+    DOUBLE = 64
+
+
 class Backend(ABC):
     """Abstract base class defining the interface for autodiff backends."""
 
     ##########################################################################
     # Common attributes, properties and methods
     ##########################################################################
     _dtype: type
+    _dtype_precision: DTypePrecision
 
-    @runtime_checkable
+    @runtime_checkable  # to be able to check isinstance(x, bk.array_t)
```

Comment:
why did you add this comment?

Reply:
I felt I needed to explain what 'checkable' meant in this context so that devs would not need to read the python docu. Happy to remove it if you judge it unnecessary